src/primitives/bezier.jl

```diff
@@ -0,0 +1,89 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    BezierCurve(points)
+
+A Bézier curve with control points `points`.
+See https://en.wikipedia.org/wiki/Bézier_curve.
+A point on the curve `b` can be evaluated by calling
+`b(t)` with `t` between `0` and `1`.
+The evaluation method defaults to DeCasteljau's algorithm
+for accurate evaluation. Horner's method, faster with a
+large number of points but less precise, can be used via
+`b(t, Horner())`.
+
+## Example
+
+```julia
+BezierCurve(Point2[(0.,0.),(1.,-1.)])
+```
+"""
+struct BezierCurve{Dim,T,V<:AbstractVector{Point{Dim,T}}} <: Primitive{Dim,T}
+  controls::V
+end
+
+BezierCurve(points::Vararg) = BezierCurve(collect(points))
+BezierCurve(points::AbstractVector{<:Tuple}) = BezierCurve(Point.(points))
+
+ncontrols(b::BezierCurve) = length(b.controls)
+
+"""
+Evaluation method used to obtain a point along
+a Bézier curve from a parametric expression.
+"""
+abstract type BezierEvalMethod end
```

Comment:
This abstract type is not concretely used, but I found it was clearer to appropriately subtype both method types.